src/shared/services/UnreadCounterService.ts

```diff
@@ -0,0 +1,95 @@
+import { UserService, HttpService } from "../services";
+import { updateUnreadCountsInterval } from "../config";
+import { poll } from "@utils/helpers";
+import { myAuth } from "@utils/app";
+import { amAdmin } from "@utils/roles";
+import { isBrowser } from "@utils/browser";
+
+/**
+ * Service to poll and keep track of unread messages / notifications.
+ */
+export class UnreadCounterService {
+  // fetched by HttpService.getUnreadCount, appear in inbox
+  unreadPrivateMessages = 0;
+  unreadReplies = 0;
+  unreadMentions = 0;
+  public unreadInboxCount = 0;
+
+  // fetched by HttpService.getReportCount, appear in report page
+  commentReportCount = 0;
+  postReportCount = 0;
+  messageReportCount = 0;
+  public unreadReportCount = 0;
+
+  // fetched by HttpService.getUnreadRegistrationApplicationCount, appear in registration application page
+  public unreadApplicationCount = 0;
+
+  static #instance: UnreadCounterService;
+  private subscriptions = new Set<(service: UnreadCounterService) => any>();
```

Comment:
A bit of a hack right now.
A more "react" way to do this would be to use React.createContext and React.useContext for a global state.
This here has the advantage of not requiring to be called from a react component

Reply:
Inferno does have support for contexts, but they are undocumented and have the same API as contexts for react class components. The only point I have is on the type: will the functions in the set actually return anything? If not, you could swap the `any` for `void`. If it does have to return something, you could try:
```typescript
<T>(service: UnreadCounterService) => T
```